### PR TITLE
Add CI on riscv-arch-test based on mill

### PR DESCRIPTION
**Pull request by ZenithalHourlyRate**
**Related issue**: #3191, #3192, #3193

Cherry-picked commits from them to test CI. Especially, #3193 is for passing ebreak test. These commits will be removed once they have been merged. The successful running of CI introduced by this PR may indicate that these PR could be merged.

This CI may finish running in a reasonable time. In my machine, running arch-test without floating point finished in one hour.

**Type of change**: other enhancement

**Impact**: no functional change

**Development Phase**: implementation

**Release Notes**

Add new CI on riscv-arch-test based on mill

### CODE REVIEW

**Comment by ZenithalHourlyRate** [comment]
Unfortunately still hit OOM

**Comment by ZenithalHourlyRate** [comment]
Finally finished running!

> Unfortunately still hit OOM

It turned out I did not set the correct timeout for emulators. It worked on my machine because I have a large parallism so timeout did not function correctly.

**Comment by ZenithalHourlyRate** [comment]
After #3191, this can be merged.

Since there are ebreak related issues #3193, we can disable privileged tests (thus zicsr) for now so that this CI could run. This will unblock #2950.

**Review by sequencer — APPROVED** [review]

**Pull request merged**
